game/game.py

```diff
@@ -437,16 +437,48 @@ def plan_procurement(self) -> None:
         # gets much more of the budget that turn. Otherwise budget (after
         # repairs) is split evenly between air and ground. For the default
         # starting budget of 2000 this gives 600 to ground forces and 1400 to
-        # aircraft.
-        ground_portion = 0.3 if self.turn == 0 else 0.5
+        # aircraft. After that the budget will be spend proportionally based on how much is already invested
+
+        player_armor_investment = 0
+        player_aircraft_investment = 0
+        player_ground_portion = 0.3
+
+        if self.turn is not 0:
+            for cp in self.theater.player_points():
+                player_armor_investment += cp.base.total_ground_units_value
+                player_aircraft_investment += cp.base.total_aircraft_value
+
+            player_combined_unit_value = (
+                player_aircraft_investment + player_armor_investment
+            )
+            player_ground_portion = (
+                player_aircraft_investment / player_combined_unit_value
+            )
+
+        enemy_armor_investment = 0
+        enemy_aircraft_investment = 0
+        enemy_ground_portion = 0.3
+
+        if self.turn is not 0:
+            for cp in self.theater.enemy_points():
+                enemy_armor_investment += cp.base.total_ground_units_value
+                enemy_aircraft_investment += cp.base.total_aircraft_value
+
+            enemy_combined_unit_value = (
+                enemy_aircraft_investment + enemy_armor_investment
+            )
+            enemy_ground_portion = (
+                player_aircraft_investment / enemy_combined_unit_value
+            )
```

Comment:
Dedup all of this into the procurer itself? It doesn't _need_ to be an argument to the constructor, it can do this calculation so you don't have to copy paste it for red and blue.

Reply:
done